lib/cl/conditions.lisp

```diff
@@ -0,0 +1,263 @@
+(in-package :org.apache.thrift.implementation)
+
+;;; This file defines exception classes and signaling operators for the `org.apache.thrift` library.
+;;;
+;;; copyright 2010 [james anderson](james.anderson@setf.de)
+;;;
+;;; Licensed to the Apache Software Foundation (ASF) under one
+;;; or more contributor license agreements. See the NOTICE file
+;;; distributed with this work for additional information
+;;; regarding copyright ownership. The ASF licenses this file
+;;; to you under the Apache License, Version 2.0 (the
+;;; "License"); you may not use this file except in compliance
+;;; with the License. You may obtain a copy of the License at
+;;; 
+;;;   http://www.apache.org/licenses/LICENSE-2.0
+;;; 
+;;; Unless required by applicable law or agreed to in writing,
+;;; software distributed under the License is distributed on an
+;;; "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+;;; KIND, either express or implied. See the License for the
+;;; specific language governing permissions and limitations
+;;; under the License.
+
+
+;;; The exception hierarchy is rooted in thrift-error and mixes in other standard
+;;; conditions as appropriate for the excpetion attributes
+;;;
+;;; thrift-error
+;;; - application-error
+;;; - protocol-error
+;;;   - protocol-version-error
+;;;   - protocol-type-error (type-error)
+;;; - unknown-field-error (cell-error)
+;;; - field-type-error (type-error)
+;;; - transport-error
+;;; 
+
+
+;;; abstract exceptions
+
+(define-condition thrift-error (simple-error)
+  ((type
+    :initform *protocol-ex-unknown*
+    :reader thrift-error-type))
+  (:report (lambda (error stream)
+             (apply #'format stream (thrift-error-format-control error)
+                    (thrift-error-format-arguments error)))))
+
+(defgeneric thrift-error-format-control (error)
+  (:method ((error thrift-error))
+    "~(~a~): ~a."))
+
+(defgeneric thrift-error-format-arguments (error)
+  (:method ((error thrift-error))
+    (list (type-of error) (thrift-error-type error))))
+
+
+
+(define-condition protocol-error (thrift-error)
+  ((protocol :initarg :protocol :initform nil
+             :reader protocol-error-protocol)))
+
+(defmethod thrift-error-format-control ((error protocol-error))
+  (concatenate 'string (call-next-method)
+               " connection: ~a."))
+
+(defmethod thrift-error-format-arguments ((error protocol-error))
+  (append (call-next-method)
+          (list (protocol-error-protocol error))))
+
+
```

Comment:
cosmetic: one newline between definitions (if no comment inbetween) is more than enough. remove such gaps.